programs/storage_api/src/storage_instruction.rs

```diff
@@ -27,12 +40,71 @@ pub enum StorageInstruction {
     },
 }
 
-pub fn create_account(from: &Pubkey, to: &Pubkey, lamports: u64) -> Instruction {
-    system_instruction::create_account(&from, to, lamports, STORAGE_ACCOUNT_SPACE, &id())
+pub fn create_validator_storage_account(
+    from_pubkey: &Pubkey,
+    storage_pubkey: &Pubkey,
+    lamports: u64,
+) -> Vec<Instruction> {
+    vec![
+        system_instruction::create_account(
+            from_pubkey,
+            storage_pubkey,
+            lamports,
+            STORAGE_ACCOUNT_SPACE,
+            &id(),
+        ),
+        Instruction::new(
+            id(),
+            &StorageInstruction::InitializeValidatorStorage,
+            vec![AccountMeta::new(*storage_pubkey, false)],
+        ),
+    ]
 }
 
-pub fn mining_proof(
+pub fn create_replicator_storage_account(
     from_pubkey: &Pubkey,
+    storage_pubkey: &Pubkey,
+    lamports: u64,
+) -> Vec<Instruction> {
+    vec![
+        system_instruction::create_account(
+            from_pubkey,
+            storage_pubkey,
+            lamports,
+            STORAGE_ACCOUNT_SPACE,
+            &id(),
+        ),
+        Instruction::new(
+            id(),
+            &StorageInstruction::InitializeReplicatorStorage,
+            vec![AccountMeta::new(*storage_pubkey, false)],
+        ),
+    ]
+}
+
+pub fn create_mining_pool_account(
+    from_pubkey: &Pubkey,
+    storage_pubkey: &Pubkey,
+    lamports: u64,
+) -> Vec<Instruction> {
+    vec![
+        system_instruction::create_account(
+            from_pubkey,
+            storage_pubkey,
+            lamports,
+            STORAGE_ACCOUNT_SPACE,
+            &id(),
+        ),
+        Instruction::new(
+            id(),
+            &StorageInstruction::InitializeMiningPool,
+            vec![AccountMeta::new(*storage_pubkey, false)],
+        ),
+    ]
+}
+
+pub fn mining_proof(
+    storage_pubkey: &Pubkey,
```

Comment:
does this TX need a from_pubkey for #3880 ?

Reply:
I'm gonna make storage use a system account for fees immediately after I'm done with this PR